src/Management/src/EndpointCore/ActuatorRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,147 @@
+                case Type _ when endpointType.IsAssignableFrom(typeof(RefreshEndpoint)):
+                    return (typeof(RefreshEndpointMiddleware), typeof(IRefreshOptions));
+                case Type _ when endpointType.IsAssignableFrom(typeof(ThreadDumpEndpoint)):
+                    return (typeof(ThreadDumpEndpointMiddleware), typeof(IThreadDumpOptions));
+                case Type _ when endpointType.IsAssignableFrom(typeof(ThreadDumpEndpoint_v2)):
+                    return (typeof(ThreadDumpEndpointMiddleware_v2), typeof(IThreadDumpOptions));
+                case Type _ when endpointType.IsAssignableFrom(typeof(TraceEndpoint)):
+                    return (typeof(TraceEndpointMiddleware), typeof(ITraceOptions));
+                case Type _ when endpointType.IsAssignableFrom(typeof(HttpTraceEndpoint)):
+                    return (typeof(HttpTraceEndpointMiddleware), typeof(ITraceOptions));
+                case Type _ when endpointType.IsAssignableFrom(typeof(CloudFoundryEndpoint)):
+                    return (typeof(CloudFoundryEndpointMiddleware), typeof(ICloudFoundryOptions));
+            }
+
+            throw new InvalidOperationException($"Could not find middleware for Type: {endpointType.Name} ");
+        }
+
+        /// <summary>
+        /// Generic routebuilder extension for Actuators.
+        /// </summary>
+        /// <param name="endpoints">IEndpointRouteBuilder to Map route.</param>
+        /// <typeparam name="TEndpoint">Middleware for which the route is mapped.</typeparam>
+        /// <exception cref="InvalidOperationException">When T is not found in service container</exception>
+        public static void Map<TEndpoint>(this IEndpointRouteBuilder endpoints)
+        where TEndpoint : IEndpoint
+        {
+            if (endpoints == null)
+            {
+                throw new ArgumentNullException(nameof(endpoints));
+            }
+
+            var (middleware, optionsType) = LookupMiddleware(typeof(TEndpoint));
+            var options = endpoints.ServiceProvider.GetService(optionsType) as IEndpointOptions;
+            var mgmtOptionsCollection = endpoints.ServiceProvider.GetServices<IManagementOptions>();
+
+            foreach (var mgmtOptions in mgmtOptionsCollection)
+            {
+                if ((mgmtOptions is CloudFoundryManagementOptions && options is IActuatorHypermediaOptions)
+                    || (mgmtOptions is ActuatorManagementOptions && options is ICloudFoundryOptions))
+                {
+                    continue;
+                }
+
+                var fullPath = options.GetContextPath(mgmtOptions);
+
+                var pipeline = endpoints.CreateApplicationBuilder()
+                    .UseMiddleware(middleware, mgmtOptions)
+                    .Build();
+
+                if (options.AllowedVerbs == null)
+                {
+                    endpoints.Map(fullPath, pipeline);
```

Comment:
Will this result in all verbs being mapped? I'd think we'd want to limit it to `GET` by default

Reply:
You are correct - good catch!